### PR TITLE
Reset rate limit plugin

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR introduces a Poke plugin to reset the rate limiter for a given workspace. I refactored a bit the rate limiter so we can avoid magic variables.

### CODE REVIEW

File: front/lib/api/assistant/rate_limits.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  MaxMessagesTimeframeType,
+} from "@dust-tt/types";
+import { expireRateLimiterKey } from "@dust-tt/types";
+
+import type { Authenticator } from "@app/lib/auth";
+
+export const makeMessageRateLimitKeyForWorkspace = (
+  owner: LightWorkspaceType
+) => {
+  return `postUserMessage:${owner.sId}`;
```
Comment (JulesBelveze) [inline front/lib/api/assistant/rate_limits.ts:12]: `nit`: name is a bit misleading no?

Comment (JulesBelveze) [reply front/lib/api/assistant/rate_limits.ts:12]: Could we follow the same pattern as the key below?

**Review by tdraier — APPROVED** [review]

File: types/src/shared/rate_limiter.ts [hunk 2]
```diff
@@ -3,37 +3,49 @@ import { v4 as uuidv4 } from "uuid";
 import { MaxMessagesTimeframeType } from "../front/plan";
 import { LoggerInterface } from "../shared/logger";
 import { redisClient } from "../shared/redis_client";
+import { Err, Ok, Result } from "./result";
 import { getStatsDClient } from "./statsd";
 
 export class RateLimitError extends Error {}
 
+async function getRedisClient(redisUri?: string) {
```
Comment (JulesBelveze) [inline types/src/shared/rate_limiter.ts:11]: IIRC we already have such function right, could we reuse it ? Or move the existing one to `types`?

File: types/src/shared/rate_limiter.ts [hunk 3]
```diff
@@ -80,6 +92,28 @@ export async function rateLimiter({
...
 
+export async function expireRateLimiterKey({
+  key,
+  redisUri,
+}: RateLimiterOptionsBase): Promise<Result<boolean, Error>> {
+  let redis: undefined | Awaited<ReturnType<typeof redisClient>> = undefined;
+
+  try {
+    redis = await getRedisClient(redisUri);
+    const redisKey = makeRateLimiterKey(key);
```
Comment (JulesBelveze) [inline types/src/shared/rate_limiter.ts:103]: Shouldn't this be done in `makeMessageRateLimitKeyForWorkspace`?

File: front/lib/api/assistant/rate_limits.ts (hunk 1 above)
Comment (flvndvd) [reply front/lib/api/assistant/rate_limits.ts:12]: What do you mean?

File: types/src/shared/rate_limiter.ts (hunk 2 above)
Comment (flvndvd) [reply types/src/shared/rate_limiter.ts:11]: Yep follow up PR to extract redis from types.

File: types/src/shared/rate_limiter.ts (hunk 3 above)
Comment (flvndvd) [reply types/src/shared/rate_limiter.ts:103]: It adds the `rate_limiter:` prefix.

File: front/lib/api/assistant/rate_limits.ts (hunk 1 above)
Comment (JulesBelveze) [reply front/lib/api/assistant/rate_limits.ts:12]: My bad thought you created the key

File: types/src/shared/rate_limiter.ts (hunk 2 above)
Comment (JulesBelveze) [reply types/src/shared/rate_limiter.ts:11]: 🙏🏼

File: types/src/shared/rate_limiter.ts (hunk 3 above)
Comment (JulesBelveze) [reply types/src/shared/rate_limiter.ts:103]: Makes kinda sense to do it within `makeMessageRateLimitKeyForWorkspace` then, no? 😸

**Pull request merged**